### PR TITLE
Add "in_cache" feature to desktop KPI report

### PR DESCRIPTION
**Pull request by fbertsch**
This adds a new element that replaces the previous message (loads in 2-3 seconds after re-running). See below for a quick demo.

https://user-images.githubusercontent.com/20819040/114606323-23241b00-9c69-11eb-915b-023506fa31df.mov

### CODE REVIEW

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 1]
```diff
@@ -405,3 +405,30 @@ view: prediction {
...
+      {% condition firefox_desktop_usage_2021.medium %} medium {% endcondition %}
+      {% condition firefox_desktop_usage_2021.os %} os {% endcondition %}
+      {% condition firefox_desktop_usage_2021.source %} source {% endcondition %}
+      {% condition firefox_desktop_usage_2021.attributed %} (attributed) {% endcondition %}
+      """, r"(\(.*\))"), '') ;;
+  }
+
+  measure: is_cached {
+    type: string
+    sql: if(MAX(row_count) > 0, "Retreiving cached value...", "Building forecast model, will take 2-3 minutes...") ;;
```
Comment (jklukas) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:432]:
```suggestion
    sql: if(MAX(row_count) > 0, "Retrieving cached value...", "Building forecast model, will take 2-3 minutes...") ;;
```

I before e, except after c

Or when sounded as 'a' as in 'neighbor' and 'weigh'

Unless the 'c' is part of a 'sh' sound as in 'glacier'

Or it appears in comparatives and superlatives like 'fancier'

And also except when the vowels are sounded as 'e' as in 'seize'

Or 'i' as in 'height'

Or also in '-ing' inflections ending in '-e' as in 'cueing'

Or in compound words as in 'albeit'

Or occasionally in technical words with strong etymological links to their parent languages as in 'cuneiform'

Or in other numerous and random exceptions such as 'science', 'forfeit', and 'weird'.

- [Merriam-Webster](https://www.merriam-webster.com/words-at-play/i-before-e-except-after-c)

**Comment by jklukas** [comment]
This is cool. Does the message persist after the page has finished loading?

**Comment by fbertsch** [comment]
> This is cool. Does the message persist after the page has finished loading?

It does. I didn't experiment with trying to figure out if the dashboard is loaded, I can't think of any Looker capabilities for that :/

**Review by irrationalagent — APPROVED** [review]
super clever, thanks for doing this frank it really improves the usability.

**Comment by fbertsch** [comment]
Added a refresh. Now when the cache is filled it will switch to the other text. Should we change it to something like `Slice is cached`? cc @irrationalagent @jklukas

**Comment by irrationalagent** [comment]
> Added a refresh. Now when the cache is filled it will switch to the other text. Should we change it to something like `Slice is cached`? cc @irrationalagent @jklukas

Maybe "The forecast for the current selection has now been cached for future use?" or is that too much verbiage? Trying to think of a less technical term than "slice"

**Comment by fbertsch** [comment]
Okay, I updated it to `The forecast for the current selection is cached.`. That should make it obvious what the difference is.

**Pull request merged**